src/command/supergraph/compose/do_compose.rs

```diff
@@ -187,33 +201,60 @@ impl Compose {
             &federation_version
         );
 
-        let output = Command::new(&exe)
-            .args(["compose", yaml_path.as_ref()])
-            .output()
-            .context("Failed to execute command")?;
-        let stdout = str::from_utf8(&output.stdout)
-            .with_context(|| format!("Could not parse output of `{} compose`", &exe))?;
-
-        match serde_json::from_str::<BuildResult>(stdout) {
-            Ok(build_result) => match build_result {
-                Ok(build_output) => Ok(CompositionOutput {
-                    hints: build_output.hints,
-                    supergraph_sdl: build_output.supergraph_sdl,
-                    federation_version: Some(federation_version.to_string()),
-                }),
-                Err(build_errors) => Err(RoverError::from(RoverClientError::BuildErrors {
-                    source: build_errors,
-                    num_subgraphs,
-                })),
-            },
-            Err(bad_json) => Err(anyhow!("{}", bad_json))
-                .with_context(|| anyhow!("{} compose output: {}", &exe, stdout))
-                .with_context(|| anyhow!("Output from `{} compose` was malformed.", &exe))
-                .map_err(|e| {
-                    let mut error = RoverError::new(e);
-                    error.set_suggestion(RoverErrorSuggestion::SubmitIssue);
-                    error
-                }),
+        // Whether we use stdout or a file dependson whether the the `--output` option was used
+        let content = match output_file {
+            // If it was, we use a file in the supergraph binary; this cuts down the overall time
+            // it takes to do composition when we're working on really large compositions, but it
+            // carries with it the assumption that stdout is superfluous
+            Some(filepath) => {
+                Command::new(&exe)
+                    .args(["compose", yaml_path.as_ref(), filepath.as_ref()])
+                    .output()
+                    .context("Failed to execute command")?;
```

Comment:
We should check the version of composition and return a useful error if the user tries to use the new option with a too-old version of composition. Otherwise, they'll get a confusing message, since it's really not on them to know that we're passing through the option to another binary.

In fact, it might be even better to support the new option for older compositions, and we just redirect the output ourselves instead of passing through the option?

Reply:
ahh, smart! I added a check for supergraph binary version and am ignoring the flag when it's lower than 2.9.0

since this isn't a customer-requested feature, I'm going to punt on supporting older compositions by adding more logic--we can always add it later if/when folks want it